src/datascience-ui/native-editor/redux/reducers/creation.ts

```diff
@@ -100,8 +100,8 @@ export namespace Creation {
         if (position >= 0) {
             newList.splice(position, 0, newVM);
         } else {
-            newList.splice(0, 0, newVM);
-            position = 0;
+            newList.splice(newList.length, 0, newVM);
```

Comment:
This fixes the problem in the custom editor with undo. When the 'above' cell isn't found it was inserting at the top instead of the bottom.

Reply:
@rchiodo Sorry wasn't fully getting this. insertAbove requires the vm value, right? Wouldn't the case where we can't find that vm be an error? Or if it's not an error I'm not sure why the default for insertAbove would be the end and not the start.

Reply:
It doesn't require the VM value, but it did assume if it wasn't found it should insert at the top. The way the function is called though is it passes an empty value when it expects it to be inserted at the end.

I agree it's confusing but it was that or special case the delete to have a new insert. Maybe that would be better. It'd certainly make more sense given the name and the common assumption that it would insert above everything if it can't find the item to insert above.

Reply:
But I can also see the argument that if I can't find what to insert above, why would I assume above everything? If there's an item in the list I should definitely be able to find it.

Reply:
Ok, I get it a bit more now. The API is just a bit wonky for that function. I think that special casing the delete might be better if we are getting super strict. But at this point, especially with us moving away from the webviews, I'm perfectly fine with trusting your judgement on if that's worth the time.

Reply:
So I'll approve.

Reply:
Thanks. I added a separate code path for delete outside the index.